File: 1675-minimize-deviation-in-array/1675-minimize-deviation-in-array.cpp
class Solution {
public:
    int minimumDeviation(vector<int>& nums) {
     int mn=INT_MAX;
        int res=INT_MAX;
        priority_queue<int>q;
        for(int i=0;i<nums.size();i++){
            if(nums[i]%2!=0)nums[i]=nums[i]*2;
            mn=min(nums[i],mn);
            q.push(nums[i]);
        }
        while(q.top()%2==0){
            res=min(res,q.top()-mn);
            q.push(q.top()/2);
            mn=min(q.top()/2,mn);
            q.pop();
        }
   return res=min(res,q.top()-mn);     
        
        
        
    }
};

File: 1675-minimize-deviation-in-array/1675-minimize-deviation-in-array.java
class Solution {
    public int minimumDeviation(int[] nums) {
        int mn=Integer.MAX_VALUE,res=Integer.MAX_VALUE;
        PriorityQueue<Integer>pq=new PriorityQueue<>(Collections.reverseOrder());
        for(int i:nums){
            if(i%2!=0)i*=2;
            pq.add(i);
            mn=Math.min(mn,i);
        }
        while(pq.peek()%2==0){
            int temp=pq.poll();
            res=Math.min(res,temp-mn);
            pq.add(temp/2);
            mn=Math.min(mn,temp/2);
        }
        return Math.min(res,pq.peek()-mn);
    }
}